lib/trento/application/integration/discovery/policies/sap_system_policy.ex

```diff
@@ -166,9 +175,23 @@ defmodule Trento.Integration.Discovery.SapSystemPolicy do
 
   @spec parse_instance_number(map) :: String.t()
   defp parse_instance_number(%{
-         "SAPControl" => %{"Properties" => %{"SAPSYSTEM" => %{"value" => instance_number}}}
-       }),
-       do: instance_number
+         "SAPControl" => %{"Properties" => properties}
+       })
+       do
+
+       properties
+       |> Enum.find(fn %{"property" => property} ->
+         property == "SAPSYSTEM"
+       end)
+       |> case do
+         %{"value" => value} ->
+           {:ok, value}
+
+         _ ->
+           {:error, :key_not_found}
```

Comment:
by using find value we can return value or nil here instead of the ok/error tuple